src/main/java/com/solace/samples/jms/snippets/HowToEnablePayloadCompression.java

```diff
@@ -0,0 +1,148 @@
+     *
+     * @see SupportedProperty#SOLACE_JMS_PAYLOAD_COMPRESSION_LEVEL
+     */
+    public static void sendCompressedBytesMessage(String jndiProviderURL,String username, int payloadCompressionLevel) throws NamingException, JMSException {
+        InitialContext initialContext = createInitialContextWithPayloadCompression(jndiProviderURL, username, payloadCompressionLevel);
+
+        ConnectionFactory cf = (ConnectionFactory) initialContext.lookup(CONNECTION_FACTORY_JNDI_NAME);
+        Connection connection = cf.createConnection();
+        Session session = connection.createSession(false, Session.AUTO_ACKNOWLEDGE);
+        Topic topic = session.createTopic(TOPIC_NAME);
+        MessageProducer producer = session.createProducer(topic);
+
+        BytesMessage msg = session.createBytesMessage();
+        msg.writeBytes(getCompressibleMessage().getBytes());
+
+        producer.send(msg);
+    }
+
+    /**
+     * Sample of how to enable compressed payload to send a {@code TextMessage}.
+     *
+     * @param jndiProviderURL The URL to the solace broker.
+     * @param username The username to connect to the solace broker.
+     * @param payloadCompressionLevel A compression level the range 0-9.
+     *     <p>Value meanings:
+     *     <ul>
+     *       <li>0 - disable payload compression (the default)
+     *       <li>1 - least amount of compression and fastest data throughput
+     *       <li>9 - most compression and slowest data throughput
+     *     </ul>
+     *
+     * @see SupportedProperty#SOLACE_JMS_PAYLOAD_COMPRESSION_LEVEL
+     */
+    public static void sendCompressedTextMessage(String jndiProviderURL, String username, int payloadCompressionLevel) throws NamingException, JMSException {
+        InitialContext initialContext = createInitialContextWithPayloadCompression(jndiProviderURL, username, payloadCompressionLevel);
+
+        ConnectionFactory cf = (ConnectionFactory) initialContext.lookup(CONNECTION_FACTORY_JNDI_NAME);
+        Connection connection = cf.createConnection();
+        Session session = connection.createSession(false, Session.AUTO_ACKNOWLEDGE);
+        Topic topic = session.createTopic(TOPIC_NAME);
+        MessageProducer producer = session.createProducer(topic);
+
+        TextMessage msg = session.createTextMessage(getCompressibleMessage());
+
+        /*
+         * Note: that SOLACE_JMS_PROP_ISXML needs to be false for payload compression to occur. This is only needed for TextMessage.
+         * No other message type need this property enabled.
+         *
+         * Alternatively, in the jndi connection factory on the broker set "Publisher Text Messages as XML Payload" to false (Web interface).
+         * This will have the same behaviour.
+         */
+        msg.setBooleanProperty(SupportedProperty.SOLACE_JMS_PROP_ISXML, false);
+
+        producer.send(msg);
+    }
+
+    private static String getCompressibleMessage() {
+        char[] msg = new char[100];
+        Arrays.fill(msg, 'A');
```

Comment:
haha I'm sure this compresses quite nicely!  (no code change here, I'm just saying)